cardinal/config.go

```diff
@@ -160,3 +189,52 @@ func (w *WorldConfig) setLogger() error {
 
 	return nil
 }
+
+func setupViper() {
+	once.Do(func() {
+		// Bind the command-line flags to Viper
+		pflag.String(tomlConfigFileEnvVariable, "", "path to the toml config file")
+		pflag.Parse()
+		err := viper.BindPFlags(pflag.CommandLine)
+		if err != nil {
+			log.Debug().Err(err).Msg("failed to bind command-line flags to Viper")
+			// Continue even if the binding fails
+		}
+	})
+
+	// Bind the environment variables to Viper
+	viper.AutomaticEnv()
+
+	// Set default toml config file name and type
+	viper.SetConfigName("world") // name of config file (without extension)
+	viper.SetConfigType("toml")  // REQUIRED if the config file does not have the extension in the name
+
+	// Find the toml file from the flag and env variable
+	tomlFilePath := viper.GetString(tomlConfigFileEnvVariable)
+	if tomlFilePath == "" {
+		// Find the toml file in the current and parent directories
+		viper.AddConfigPath(".") // look for config in the working directory
+
+		// Add parent directory to the search path
+		parentDir, err := os.Getwd()
+		if err != nil {
+			log.Warn().Err(err).Msg("failed to get current directory for toml file search")
+		} else {
+			parentDir = filepath.Dir(parentDir)
+			viper.AddConfigPath(parentDir)
+		}
+	} else {
+		// Use the toml file from the flag or env variable
```

Comment:
the priority for the config path should be flag -> env -> default path (world.toml in current directory or parent's directory)

Reply:
flag and env precedence is handled by viper. it should be already like flag > env > default (reference : https://github.com/spf13/viper#why-viper).
if flag and env is not set then we search in current dir and parrent dir.